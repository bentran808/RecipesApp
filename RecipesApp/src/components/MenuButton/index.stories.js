import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import CenterView from 'storybook/stories/CenterView';
import MenuButton from '.';

storiesOf('Menu Button', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('default', () => <MenuButton onPress={action('clicked-icon')} />);
