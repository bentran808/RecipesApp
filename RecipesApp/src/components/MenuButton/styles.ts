import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  headerButtonContainer: {
    padding: 10
  },
  headerButtonImage: {
    justifyContent: 'center',
    width: 25,
    height: 25,
    margin: 6
  },
  badge: {
    position: 'absolute',
    right: 10,
    top: 10,
    backgroundColor: 'red',
    borderRadius: 8,
    width: 16,
    height: 16,
    justifyContent: 'center',
    alignItems: 'center'
  }
});

export default styles;
