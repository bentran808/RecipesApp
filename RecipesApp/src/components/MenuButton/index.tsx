import React from 'react';
import { Image, TouchableOpacity } from 'react-native';
import { MenuIcon } from 'theme';
import styles from './styles';

type Props = {
    onPress: () => void
}

const MenuButton = ({ onPress }: Props) => {
  return (
    <TouchableOpacity style={styles.headerButtonContainer} onPress={onPress}>
      <Image style={styles.headerButtonImage} source={MenuIcon} />
    </TouchableOpacity>
  )
}

export default React.memo(MenuButton);