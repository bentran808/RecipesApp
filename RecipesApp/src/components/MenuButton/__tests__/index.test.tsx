import MenuButton from 'components/MenuButton';
import renderer from 'react-test-renderer';

describe('Menu Button Component', () => {
  test('should render correctly', () => {
    const tree = renderer.create(<MenuButton onPress={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
