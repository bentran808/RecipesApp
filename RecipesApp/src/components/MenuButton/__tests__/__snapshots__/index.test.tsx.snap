// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Button Component should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 25,
        "justifyContent": "center",
        "margin": 6,
        "width": 25,
      }
    }
  />
</View>
`;
